pkg/packageinstall/packageinstall.go

```diff
@@ -239,7 +295,31 @@ func (pi *PackageInstallCR) referencedPkgVersion() (datapkgingv1alpha1.Package,
 
 	verConfig := verv1alpha1.VersionSelection{Semver: semverConfig}
 
-	selectedVersion, err := versions.HighestConstrainedVersion(versionStrs, verConfig)
+	kcConstraint := func(pkgVer string) bool {
+		pkg := versionToPkg[pkgVer]
+		return pi.kcVersionConstraintsSatisfied(&pkg)
+	}
+
+	vi, err := pi.deployFactory.GetClusterVersion(pi.model.Spec.ServiceAccountName, pi.model.Spec.Cluster, &pi.model.ObjectMeta, pi.log)
```

Comment:
Do we make the tradeoff of a two-pass filter vs a call to the cluster for k8s info on each reconcile....

By two pass I mean: first pass --> check if any packageinstall has constraints (if not, skip over this code), second pass as is currently.

Reply:
or it can also be memoized so that we get the best of both worlds.

Reply:
Maybe it can be in the case of the on-cluster version. But should we really "cache" the version for an external cluster? Wouldn't that be prone to stale-ness?